ubo-common/src/main/java/org/mycore/ubo/resources/UBOPersonSearchResource.java

```diff
@@ -18,45 +18,51 @@
 
 package org.mycore.ubo.resources;
 
-import com.google.gson.Gson;
-import org.mycore.access.MCRAccessManager;
-import org.mycore.common.MCRSession;
-import org.mycore.common.MCRSessionMgr;
-import org.mycore.common.MCRSystemUserInformation;
-import org.mycore.common.MCRUserInformation;
-import org.mycore.common.config.MCRConfiguration2;
-import org.mycore.ubo.picker.IdentityService;
-import org.mycore.ubo.picker.PersonSearchResult;
-
 import javax.naming.OperationNotSupportedException;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
 
+import com.google.gson.Gson;
+import org.mycore.access.MCRAccessManager;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.ubo.local.LocalService;
+import org.mycore.ubo.picker.IdentityService;
+import org.mycore.ubo.picker.PersonSearchResult;
+
 @Path(UBOPersonSearchResource.PATH)
 public class UBOPersonSearchResource {
 
     private static final String STRATEGY_CONFIG_STRING = "MCR.IdentityPicker.strategy";
 
     private static final String STRATEGY_CLASS_SUFFIX = "Service";
+
     public static final String PATH = "search/person";
 
+    private static final boolean INCLUDE_LOCAL_USERS = MCRConfiguration2.getBoolean(
+        "UBO.Search.include.Local.Users").orElse(false);
+
     @GET
     public Response search(@QueryParam("query") String searchQuery) {
         String id = "webpage:/rsc/" + PATH;
-        if(MCRAccessManager.hasRule(id, "read")){
-            if(!MCRAccessManager.checkPermission(id, "read")){
+        if (MCRAccessManager.hasRule(id, "read")) {
+            if (!MCRAccessManager.checkPermission(id, "read")) {
                 return Response.status(Response.Status.FORBIDDEN).build();
             }
         }
 
         String classPrefix = MCRConfiguration2.getStringOrThrow(STRATEGY_CONFIG_STRING);
-        IdentityService service = (IdentityService) MCRConfiguration2.instantiateClass(classPrefix + STRATEGY_CLASS_SUFFIX);
-
+        IdentityService service = MCRConfiguration2.instantiateClass(classPrefix + STRATEGY_CLASS_SUFFIX);
 
         try {
             PersonSearchResult results = service.searchPerson(searchQuery);
+            if (INCLUDE_LOCAL_USERS && !LocalService.class.getName().equals(classPrefix + STRATEGY_CLASS_SUFFIX)) {
```

Comment:
I think it would be better to move that code to an extra IdentityService. 
The SearchResource should not contain any logic it should only deliver the results from the service as json. 
The decision if the email should be send to the client is also done in the service, not in the resource.

The IdentityService could also merge the PersonSearchResults to prevent duplicate Results.

Reply:
Added extra IdentityService as requested.